mllib-dal/src/main/scala/org/apache/spark/ml/stat/CorrelationDALImpl.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.stat
+
+import java.util.Arrays
+import org.apache.spark.TaskContext
+import org.apache.spark.internal.Logging
+import org.apache.spark.ml.util.{OneCCL, OneDAL, Utils}
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.util.Utils.getOneCCLIPPort
+import org.apache.spark.rdd.RDD
+import org.apache.spark.mllib.linalg.{Matrix}
+
+class CorrelationDALImpl(
+                          val executorNum: Int,
+                          val executorCores: Int)
+  extends Serializable with Logging {
+
+  def computeCorrelationMatrix(data: RDD[Vector]): Matrix = {
+
+    val kvsIPPort = getOneCCLIPPort(data)
+
+      val sparkContext = data.sparkContext
+      val useGPU = sparkContext.conf.getBoolean("spark.oap.mllib.useGPU", false)
+
+      val coalescedTables = OneDAL.rddVectorToMergedTables(data, executorNum)
```

Comment:
need to make sure the data is cached here, otherwise ExecutorInProcessCoalescePartitioner will not work